### DISCUSSION TITLE
What is `pnpm-global` folder?

### DISCUSSION THREAD

**Discussion by miqdad-f** [pnpm/pnpm#4491] [Q&A]
Hi, quick question. So after running `pnpm --global <pkg>` pnpm apparently generates `pnpm-global` folder in my repo which contains a `node_modules/lock.yaml` and the new `pnpm-lock.yaml` inside of it.

So what actually this folder do? Does it only for "linking" purpose? And Is it safe to put the folder entirely to `.gitignore`?

**Comment by Aziks0** [comment]
If I'm not mistaken, the `pnpm-global` folder shouldn't be in any of your repos. The default path is `<path to node>/pnpm-global` as stated [here](https://pnpm.io/npmrc#global-dir). If it is generated inside one of your repos, you should fill an issue. You should also check (before filling an issue) your config using the `pnpm config get global-dir` command, to see if there is something odd.

As the name suggests, it's a folder containing the global packages you install, which basically gives you the ability to use those packages anywhere on your system, just like the `npm install --global <pkg>` command does. You should take a look at the [npm doc on the subject](https://docs.npmjs.com/downloading-and-installing-packages-globally).

And, again, because it shouldn't be in any of your repos, there should be no need to put it in a `.gitignore` file. It's entirely safe to do so though.
